include/urcu/rculfhash.h

```diff
@@ -68,6 +69,17 @@ struct cds_lfht_iter {
 #endif
 };
 
+/* cds_lfht_alloc: Callbacks if we want to use custom memory allocator. */
+struct cds_lfht_alloc {
```

Comment:
I would be more inclined if a `void *state` pointer was added to the struct.  This pointer could then be passed to the allocator interface along the other parameters.  Also, while introducing a new public API, why not add more than the basic malloc/calloc/free?  Even though, it seems that only these are currently required by the implementation, it might be nice to have the equivalent of `posix_memalign` for aligned allocations and `realloc`.

It is also important to specify in the public API documentation that an allocator _must_ implement all of the interface.

Finally, I'm against adding a `static cds_lhft_default_alloc`.  Just pass `NULL` instead and liburcu will have its own private struct internally.  Or define something along `#define CDS_LFHT_DEFAULT_ALLOCATOR NULL`.

```c
struct cds_lfht_alloc {
	void *(*malloc)(void *state, size_t size);
	void *(*calloc)(void *state, size_t nmemb, size_t size);
        void *(*realloc)(void *state, void *ptr, size_t size);
	void *(*memalign)(void *state, size_t alignment, size_t size);
	void  (*free)(void *state, void *ptr);
	void   *state;
};
```

Thanks,
Olivier

Reply:
`memalign` -> `aligned_alloc`

Reply:
Thanks for the feedback Olivier. 
I have pushed changes based on your suggestions.

I am not sure what is the best place in the public API documentation to add the requirements when using a custom allocator, so I added this as a comment on top of the struct. 
Happy to move this elsewhere, if it is more appropriate.